File: sdks/php/sdk/src/Types/TextResponseConfigFormat.php
<?php

namespace Zerogpu\Types;

use Zerogpu\Core\Json\JsonSerializableType;
use Zerogpu\Core\Json\JsonProperty;

class TextResponseConfigFormat extends JsonSerializableType
{
    /**
     * @var ?string $type Response format type (e.g. `text`)
     */
    #[JsonProperty('type')]
    public ?string $type;

    /**
     * @param array{
     *   type?: ?string,
     * } $values
     */
    public function __construct(
        array $values = [],
    ) {
        $this->type = $values['type'] ?? null;
    }

    /**
     * @return string
     */
    public function __toString(): string
    {
        return $this->toJson();
    }
}

File: sdks/csharp/src/ZerogpuApi/Types/TextResponseConfigFormat.cs
using global::System.Text.Json;
using global::System.Text.Json.Serialization;
using ZerogpuApi.Core;

namespace ZerogpuApi;

[Serializable]
public record TextResponseConfigFormat : IJsonOnDeserialized
{
    [JsonExtensionData]
    private readonly IDictionary<string, JsonElement> _extensionData =
        new Dictionary<string, JsonElement>();

    /// <summary>
    /// Response format type (e.g. `text`)
    /// </summary>
    [JsonPropertyName("type")]
    public string? Type { get; set; }

    [JsonIgnore]
    public ReadOnlyAdditionalProperties AdditionalProperties { get; private set; } = new();

    void IJsonOnDeserialized.OnDeserialized() =>
        AdditionalProperties.CopyFromExtensionData(_extensionData);

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}
